File: bridge-pattern/java/src/Shape.java
package src;

// 定义抽象形状类，关联图形操作接口
public abstract class Shape {
   protected DrawTool drawTool;

   protected Shape(DrawTool drawTool) {
      this.drawTool = drawTool;
   }

   public abstract void draw(int x, int y, int radius);
}

File: bridge-pattern/ts/src/Shape.ts
import { DrawTool } from './DrawTool.js'

// 定义抽象形状类，桥接图形操作接口DrawTool
export abstract class Shape {
  drawTool: DrawTool
  constructor(drawTool: DrawTool) {
    this.drawTool = drawTool
  }

  abstract draw(x: number, y: number, radius: number): void
}
